### ISSUE TITLE
How to distinguish between invalid auth data and service unavailable with deepstream and webhook

### ISSUE THREAD

**Issue by slachtar** [deepstreamIO/deepstream.io#1138]
Hello,
I'm using a webhoot to authenticate users https://deepstream.io/tutorials/core/auth/http-webhook/
My webhook server is working great, it authenticates users against a database.

In the last days, the database went down, my users in front end see a message saying they've provided wrong username or pass. which is totally wrong, they should see a message saying "service unavailable for the moment, please try again later".

My webhook can catch the db down error, but don't know how to report this for the front end and which status code should I send and how to retrieve it in frontend DS client.

Any pointer to docs, example is very appreciated.
Regards,

**Comment by slachtar** [comment]
Hello,
thanks for the hints, I applied the config for my DS server.
In the logs of DS, I can see the error coming:
```
Jan 25 16:40:14 testsrv deepstream[395783]: INCOMING_CONNECTION | from undefined (127.0.0.1)
Jan 25 16:40:14 testsrv deepstream[395783]: AUTH_ERROR | http auth server error: {"error":{"errorno":1,"details":"db error"}}
Jan 25 16:40:14 testsrv deepstream[395783]: AUTH_UNSUCCESSFUL | invalid authentication data
```
But the AUTH_ERROR  is never reported  on the client, all what I get in the login function is:
success : false 
clientData : { reason: 'INVALID_AUTHENTICATION_DETAILS' }

I'm using success true/false to know if the credentials are valid and show invalid login credentials on my front end in case of success: false

What I want to display in front end now is "service unavailable for the moment, please try again later" 
But I don't know how to get the AUTH_ERROR to be able to display such message.

Regards,

**Comment by jaime-ez** [comment]
Ok it seems to be a bug on the client side
here: https://github.com/deepstreamIO/deepstream.io-client-js/blob/4320d058e34886a5422c4b18bc239c2b62c8bc49/src/connection/connection.ts#L575
The message.data should be passed to onAuthUnsuccessfull in order to receive the server response. Instead it has the prewritten message that you get.

I'm kind of busy now, If you can try to solve it it would be great it should be something like redefining `onAuthUnSuccessful` to accept an argument message and pass the `message.parsedData` as argument so that value (if defined) is assigned to the reason variable.

Best

**Comment by slachtar** [comment]
Thanks, I'll look into it.

**Comment by jaime-ez** [comment]
check client v7.0.4
reopen if problem persists on client repo please

**Comment by jaime-ez** [comment]
server is not sending the right message

**Comment by jaime-ez** [comment]
solved in server v7.0.9
